cmd/kubeadm/app/apis/kubeadm/v1alpha1/defaults.go

```diff
@@ -162,7 +163,12 @@ func SetDefaults_KubeletConfiguration(obj *MasterConfiguration) {
 		obj.KubeletConfiguration.BaseConfig.AllowPrivileged = utilpointer.BoolPtr(true)
 	}
 	if obj.KubeletConfiguration.BaseConfig.ClusterDNS == nil {
-		obj.KubeletConfiguration.BaseConfig.ClusterDNS = []string{DefaultClusterDNSIP}
+		dnsIP, err := dns.GetDNSIP(obj.Networking.ServiceSubnet)
+		if err != nil {
+			obj.KubeletConfiguration.BaseConfig.ClusterDNS = []string{DefaultClusterDNSIP}
```

Comment:
In this condition, should we default it to `DefaultClusterDNSIP` or not default it a value?

Reply:
defaulting to `DefaultClusterDNSIP` is ok here